### ISSUE TITLE
BUG: Name declare cards not listing cards properly

### ISSUE THREAD

**Issue by StormWulf** [Fluorohydride/ygopro#1811]
At the start of the duel, the first declare card played works as intended as it lists every card. However, if the first declare card was successful, the next declare card will fail to search for Equip Spells, Quick-play Spells, or Normal Spells. On the opponent's turn right after, I tried playing a Mind Crush, which will fail to lookup Normal Traps and Counter Traps, but will search anything else.

Is there a possible explanation for this phenomenon? Using client version 0x1338 on Salvation Server. There are no problems at all if using LAN mode, but the issues occur when playing on the server. Would appreciate help from @IceYGO and/or @destdev.

Example of bug:
![D.D. Designator failing to declare Equip Spells](http://image.prntscr.com/image/5e0e9c226490425cbe3be7a099c080fe.png)
![D.D. Designator failing to declare Normal Spells](http://image.prntscr.com/image/c599aa9798d5482e80fd2c8a5e5f4d76.png)

**Comment by destdev** [comment]
seems like a problem on salvation only, I've just tested that on the devpro server and it works perfectly fine there ...
and no, I have no idea how that could happen ... maybe updating helps, we are at v 1.033.9 for a few months now

**Comment by Zayelion** [comment]
Salvation is using a copy of the exe binary from DevPro's latest, and we update our core every few commits. So I dont believe its something specific to Salvation. Only difference is the DB's and Salvations was recently rebuilt from the DevPro one.

**Comment by destdev** [comment]
no idea what you've made different, and if there's nothing different it seems that the cards themselves matter perhaps?
for example for my test I've used first Archfiend's Oath (resolve successfully with adding a card), then Psy-Blocker, then D.D. Designator, then Mind Crush on the start of the opponents turn ... every effect resolved properly, I was able to declare every type of card perfectly fine every time, also with testing both of your examples ("axe of" got me 3 results, "raigeki" got me 5 results in every case)

I guess you used blocker first in your examples? Well, tried that too, also works perfectly fine...

whatever it is, I fail to reproduce it on our end, sorry :/
